### PR TITLE
Remove production versions section from project guidelines

### PR DESCRIPTION
**Pull request by MarkoSagadin**
## Description

Remove production versions section from project guidelines

We never used this in practice and is a high time that it gets remove.

## Areas of interest for the reviewer

Does this change makes sense

## Checklist

- [x] My code follows the [style guidelines] as defined by IRNAS.
- [x] I have performed a self-review of my code.
- [x] My changes generate no new warnings.
- [x] I added/updated source code documentation for all newly added or changed functions.
- [x] I updated the CHANGELOG.

## After-review steps

* I will merge PR by myself.

[style guidelines]: https://github.com/IRNAS/irnas-guidelines-docs/blob/dev/docs/developer_guidelines.md

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
Current dependencies on/for this PR:
* dev
  * **PR #32** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/32" target="_blank">Graphite</a> 
    * **PR #33** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/33" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/33?utm_source=stack-comment).

**Comment by SloMusti** [comment]
@MarkoSagadin I do not agree with this as while we have not been using this yet, is something we will need in the future ans this is a good starting point.

**Comment by MarkoSagadin** [comment]
@SloMusti then I would suggest to move this section into a new document, for example `draft_ideas.md` to make it clear that we are not yet using this, but we might in the future. That way `docs/github_projects_guidelines.md` contains only things that we actively follow.

**Comment by SloMusti** [comment]
@MarkoSagadin yes, this is ok

File: CHANGELOG.md
Comment (TjazVracko) [inline CHANGELOG.md:1]: You probably formatted the file by accident. Undo?

Comment (MarkoSagadin) [reply CHANGELOG.md:1]: Reverted

**Comment by MarkoSagadin** [comment]
[Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/33) rebased this pull request as part of a merge.

**Comment by MarkoSagadin** [comment]
@MarkoSagadin merged this pull request with [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/33).

**Pull request merged**